### DISCUSSION TITLE
Please clarify launch --type=background option

### DISCUSSION THREAD

**Discussion by salmankhilji** [kovidgoyal/kitty#3600] [Q&A] [answered]
I am not completely clear as to what exactly the `--type=background` option does.  The documentation states:

> The value of background means the process will be run in the background.

So I tried to launch VS Code with the scrollback  like this:

```
map f12 launch --stdin-source=@screen_scrollback --type=background code -
```

Since VS Code is a graphical program, I want it to run in the background.  However, the above does not work.  I have to change the `type` to `overlay`, which requires Ctrl+C within the resulting window to get back to the originating window.

Is this a bug?

**Comment by kovidgoyal** [comment, marked as answer]
It means exactly what it says. the program is run as a normal program not attached to the tty device in a window, but simply in the background. stdin is passed just as normal. I dont use VS Code so I cant tell you why it might not like being launched like that.

**Reply by salmankhilji** [reply]
I tried it with `gedit` and `gvim` with no success.  The fact this issue affects 3 programs, I believe its a bug that I should file.

**Reply by salmankhilji** [reply]
For those reading this, a bug (#3602) was filed and fixed.  Thank you!
